x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorCustomSchedule.java

```diff
@@ -0,0 +1,333 @@
+            PARSER.declareBoolean(optionalConstructorArg(), SITEMAP_DISCOVERY_DISABLED_FIELD);
+            PARSER.declareStringArray(optionalConstructorArg(), DOMAIN_ALLOWLIST_FIELD);
+            PARSER.declareStringArray(optionalConstructorArg(), SITEMAP_URLS_FIELD);
+            PARSER.declareStringArray(optionalConstructorArg(), SEED_URLS_FIELD);
+        }
+
+        public static ConfigurationOverrides fromXContent(XContentParser parser) throws IOException {
+            return PARSER.parse(parser, null);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            if (maxCrawlDepth != null) {
+                builder.field(MAX_CRAWL_DEPTH_FIELD.getPreferredName(), maxCrawlDepth);
+            }
+            if (sitemapDiscoveryDisabled != null) {
+                builder.field(SITEMAP_DISCOVERY_DISABLED_FIELD.getPreferredName(), sitemapDiscoveryDisabled);
+            }
+            if (domainAllowList != null) {
+                builder.field(DOMAIN_ALLOWLIST_FIELD.getPreferredName(), domainAllowList);
+            }
+            if (sitemapUrls != null) {
+                builder.field(SITEMAP_URLS_FIELD.getPreferredName(), sitemapUrls);
+            }
+            if (seedUrls != null) {
+                builder.field(SEED_URLS_FIELD.getPreferredName(), seedUrls);
+            }
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeOptionalInt(maxCrawlDepth);
+            out.writeOptionalBoolean(sitemapDiscoveryDisabled);
+            out.writeOptionalStringCollection(domainAllowList);
+            out.writeOptionalStringCollection(sitemapUrls);
+            out.writeOptionalStringCollection(seedUrls);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            ConfigurationOverrides that = (ConfigurationOverrides) o;
+            return Objects.equals(maxCrawlDepth, that.maxCrawlDepth)
+                && Objects.equals(sitemapDiscoveryDisabled, that.sitemapDiscoveryDisabled)
+                && Objects.equals(domainAllowList, that.domainAllowList)
+                && Objects.equals(sitemapUrls, that.sitemapUrls)
+                && Objects.equals(seedUrls, that.seedUrls);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(maxCrawlDepth, sitemapDiscoveryDisabled, domainAllowList, sitemapUrls, seedUrls);
+        }
+
+        public static class Builder {
+
+            private Integer maxCrawlDepth;
+            private Boolean sitemapDiscoveryDisabled;
+            private List<String> domainAllowList;
+            private List<String> sitemapUrls;
```

Comment:
Question: Can Sitemap URLs be duplicated? Any use case for using a `Set` here instead of a `List` so we don't double our work?

Reply:
Using `Set` might be justified to deal with potential duplicates. Given that this data is used in Kibana to render user-provided input I think preserving ordering is important for UX, so I would prefer to keep it as a `List`